engine/rpcserver.go

```diff
@@ -483,38 +477,33 @@ func (s *RPCServer) GetOrderbook(ctx context.Context, r *gctrpc.GetOrderbookRequ
 		return nil, err
 	}
 
-	ob, err := s.GetSpecificOrderbook(ctx,
-		currency.Pair{
-			Delimiter: r.Pair.Delimiter,
-			Base:      currency.NewCode(r.Pair.Base),
-			Quote:     currency.NewCode(r.Pair.Quote),
-		},
-		r.Exchange,
-		a,
-	)
+	e, err := s.GetExchangeByName(r.Exchange)
 	if err != nil {
 		return nil, err
 	}
 
-	bids := make([]*gctrpc.OrderbookItem, 0, len(ob.Bids))
-	asks := make([]*gctrpc.OrderbookItem, 0, len(ob.Asks))
-	ch := make(chan bool)
+	pair := currency.Pair{
+		Delimiter: r.Pair.Delimiter,
+		Base:      currency.NewCode(r.Pair.Base),
+		Quote:     currency.NewCode(r.Pair.Quote),
+	}
+
+	ob, err := e.FetchOrderbookCached(ctx, pair, a)
+	if err != nil {
+		return nil, err
+	}
 
+	bids := make([]*gctrpc.OrderbookItem, len(ob.Bids))
+	ch := make(chan struct{})
```

Comment:
I might be being stupid, but ... 
Why is this a goro which ranges bids and just appends them?
I can't see any reason for this parallelism.

Reply:
You're right this adds in more overhead and more allocations nice catch